iOS demo app.

shoumikhin opened the pull request:
Summary:

Pre-flight:

- Clone executorch repo
- Install the recommended buck2 binary version at /tmp/buck2
- cd path/to/executorch
- Sync and update git submodules
- Make sure you have Python 3.11+ installed (standard on Ventura+) and `pip` pointing to it
- Run ./install_requirements.sh to install PyTorch dependencies

```
# Build flatbuffers compiler

cd third-party/flatbuffers && rm -rf cmake-out && mkdir cmake-out && cd cmake-out

cmake .. && cmake --build . --target flatc

cd ../../..

# Configure executorch

rm -rf cmake-out && mkdir cmake-out && cd cmake-out

cmake .. -G Xcode -DCMAKE_TOOLCHAIN_FILE=../third-party/pytorch/cmake/iOS.cmake -DBUCK2=/tmp/buck2 -DPYTHON_EXECUTABLE=$(which python3) -DFLATC_EXECUTABLE=$(realpath ../third-party/flatbuffers/cmake-out/flatc) -DCMAKE_ARCHIVE_OUTPUT_DIRECTORY=$(pwd) -DEXECUTORCH_BUILD_XNNPACK=ON -DEXECUTORCH_BUILD_EXTENSION_DATA_LOADER=ON -DCMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM=<YOUR_TEAM_ID>  # -DPLATFORM=SIMULATOR

# Build executorch

cmake --build . --config Release

# Copy executorch libs

cd Release

mkdir -p ../../examples/ios_demo_apps/ExecuTorchDemo/ExecuTorchDemo/Frameworks/executorch/

cp libclog.a libcpuinfo.a libexecutorch.a libextension_data_loader.a libportable_kernels.a libportable_ops_lib.a libpthreadpool.a libxnnpack_backend.a libXNNPACK.a ../../examples/ios_demo_apps/ExecuTorchDemo/ExecuTorchDemo/Frameworks/executorch/

cd ../..

# Download the labels and bundle with the app

mkdir -p examples/ios_demo_apps/ExecuTorchDemo/ExecuTorchDemo/Resources/Models/MobileNet/

curl https://raw.githubusercontent.com/pytorch/hub/master/imagenet_classes.txt -o examples/ios_demo_apps/ExecuTorchDemo/ExecuTorchDemo/Resources/Models/MobileNet/imagenet_classes.txt

# Export a MobileNet v3 model backed with XNNPACK delegate and copy it over to bundle with the app

export FLATC_EXECUTABLE=$(realpath third-party/flatbuffers/cmake-out/flatc)

python3 -m examples.export.export_example --model_name="mv3"
python3 -m examples.backend.xnnpack_examples --model_name="mv3" --delegate

cp mv3.pte mv3_xnnpack_fp32.pte examples/ios_demo_apps/ExecuTorchDemo/ExecuTorchDemo/Resources/Models/MobileNet/

```

Post-flight:

- Open executorch/examples/ios_demo_apps/ExecuTorchDemo/ExecuTorchDemo.xcodeproj
- Set the header search path for MobileNetClassifier target pointing to the dir containing the executorch repo
- Run the app and tests


Differential Revision: D49800418

kimishpatel reviewed (commented):
for model files bundled here, I am gonna suggest to have a python script that generates this files and link them as steps in the tutorial:
e.g.
- python generate_mv2.py --delegate...
- copy the pte file to <folder>
- ..

kimishpatel reviewed (changes requested):
sending it back for 2 things
1. if we can make model generation as part of the script that should be invoked, that would be good.
2. documentation as to how to enable your own delegate. this will help apple

shoumikhin committed:
iOS demo app. (#557)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/557

Reviewed By: cccclai

Differential Revision: D49800418

fbshipit-source-id: c4233781166453bca131c5fe0c2d76ad60f20448
